rust/src/state_machine/phases/sum2.rs

```diff
@@ -178,3 +178,105 @@ impl<R> PhaseState<R, Sum2> {
         self.mask_count() >= self.coordinator_state.min_sum_count
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::{
+        crypto::{ByteObject, EncryptKeyPair},
+        mask::{FromPrimitives, Model},
+        state_machine::{
+            coordinator::RoundSeed,
+            events::Event,
+            tests::{
+                builder::StateMachineBuilder,
+                utils::{generate_summer, generate_updater, mask_settings},
+            },
+        },
+        SumDict,
+    };
+
+    #[tokio::test]
+    pub async fn sum2_to_unmask() {
+        let n_updaters = 1;
+        let n_summers = 1;
+        let seed = RoundSeed::generate();
+        let sum_ratio = 0.5;
+        let update_ratio = 1.0;
+        let coord_keys = EncryptKeyPair::generate();
+
+        // Generate a sum dictionary with a single sum participant
+        let mut summer = generate_summer(&seed, sum_ratio, update_ratio);
+        let ephm_pk = summer.compose_sum_message(&coord_keys.public).ephm_pk();
+        let mut sum_dict = SumDict::new();
+        sum_dict.insert(summer.pk, *&ephm_pk);
```

Comment:
isn't that an identical operation resulting in the same?
```suggestion
        sum_dict.insert(summer.pk, ephm_pk);
```

Reply:
Geez yes. Sorry btw, I should have waited for your review, but I thought you were in the train.

Reply:
no worries^^. internet is lacking, but it is available from time to time.